cobalt-ast/src/cfg.rs

```diff
@@ -704,10 +756,104 @@ impl<'a, 'ctx> Cfg<'a, 'ctx> {
                 }
             }
         }
+
         errs.sort_unstable();
         errs.dedup_by_key(|m| m.loc);
+
+        let mut to_return = errs
+            .iter()
+            .map(
+                |DoubleMove {
+                     name,
+                     loc,
+                     prev,
+                     guaranteed,
+                 }| CobaltError::DoubleMove {
+                    loc: *loc,
+                    prev: *prev,
+                    name: name.clone(),
+                    guaranteed: *guaranteed,
+                },
+            )
+            .collect::<Vec<_>>();
+
+        to_return.append(&mut self.validate_linear_types(ctx));
+        to_return
+    }
+
+    fn validate_linear_types(&self, ctx: &'a CompCtx<'ctx>) -> Vec<cobalt_errors::CobaltError> {
+        let mut errs = Vec::new();
+
```

Comment:
For linear types, the condition must also be upheld before moves as well.

Reply:
What do you mean? The condition that it must be used?

Reply:
Yes. Take something like this:
```
let mut x = must_be_used();
x = must_be_used_2(); # the first value is not used!
```

Reply:
What is reassignment considered to be? A store?

Reply:
Yes. That's also the only store currently.